### ISSUE TITLE
Labels document is unreadable

### ISSUE THREAD

**Issue by morgantaschuk** [cheinle/am_production#2]
To replicate:
1. Click "Label Prep"
2. Click "Download to MS Excel worksheet"
3. Click Submit
4. Click "Click to Download File"

I get a MS Excel document with the following content:

```
慓灭敬丠浡॥瑓牡⁴慓灭楬杮䐠瑡⽥楔敭⠠奙奙䴭ⵍ䑄䠠㩈䵍区⥓倉潲敪瑣丠浡॥慓灭敬吠灹॥慓灭敬丠浵敢ॲ㈊㄰⼶㔰㈯昳物瑳瀭潲敪瑣ぁ㄰㈉㄰ⴶ㔰㈭″㜰〺㨰〰昉物瑳瀭潲敪瑣䄉〉㄰ਉ〲㘱〯⼵㌲楦獲⵴牰橯捥䉴〰ऴ〲㘱〭ⴵ㌲〠㨷〰〺र楦獲⵴牰橯捥ॴू〰ऴ㈊㄰⼶㔰㈯昴物瑳瀭潲敪瑣ぁ㈰㈉㄰ⴶ㔰㈭‴㜰〺㨰〰昉物瑳瀭潲敪瑣䄉〉㈰ਉ〲㘱〯⼵㐲楦獲⵴牰橯捥䉴〰ळ〲㘱〭ⴵ㐲〠㨷〰〺र楦獲⵴牰橯捥ॴू〰ळ㈊㄰⼶㘰㈯猱捥湯ⵤ牰橯捥䉴〰ल〲㘱〭ⴶㄲ〠㨷〰〺र敳潣摮瀭潲敪瑣䈉〉㈰ਉ〲㘱〯⼶ㄲ敳潣摮瀭潲敪瑣あ㐰㈉㄰ⴶ㘰㈭‱㜰〺㨰〰猉捥湯ⵤ牰橯捥ॴू〰ऴ㈊㄰⼶㘰㈯猱捥湯ⵤ牰橯捥䉴〰श〲㘱〭ⴶㄲ〠㨷〰〺र敳潣摮瀭潲敪瑣䈉〉㘰ਉ〲㘱〯⼶ㄲ敳潣摮瀭潲敪瑣し㄰㈉㄰ⴶ㘰㈭‱㜰〺㨰〰猉捥湯ⵤ牰橯捥ॴॗ〰ऱ㈊㄰⼶㘰㈯猱捥湯ⵤ牰橯捥坴〰ळ〲㘱〭ⴶㄲ〠㨷〰〺र敳潣摮瀭潲敪瑣圉〉㌰ਉ〲㘱〯⼶ㄲ敳潣摮瀭潲敪瑣し㔰㈉㄰ⴶ㘰㈭‱㜰〺㨰〰猉捥湯ⵤ牰橯捥ॴॗ〰व
```

**Comment by cheinle** [comment]
Hi Morgan, 

Are you opening running the browser on your ubuntu system? Does that mean you are also opening in an Office plugin of some sort or using Libre Office to open the file (also on Ubuntu or you move the file to another OS)? I am just wondering so that I can try and replicate the issue. If it is not too much trouble, might you be able to attach the xls document to this ticket? 

Have you tried downloading any other type of xls sheet from NanoLIMS? I am assuming this is also happening if you try and download an xls sheet using the Sample Query feature or the sequencing form feature?

Kind regards,

Cassie

**Comment by morgantaschuk** [comment]
Yes, the same thing happens whenever I try to download an XLS. Here's the files.

[Labels document](https://github.com/cheinle/NanoLIMS/files/415406/document_name.xls.zip)

[Samples query](https://github.com/cheinle/NanoLIMS/files/415416/samples_query.xls.zip)

**Comment by cheinle** [comment]
Hi Morgan,

I hope you are having a pleasant weekend. Sorry I have not written sooner. I have looked into this a bit and am wondering if when you are importing/opening the document that you are using the UTF-8 char set? The document should be downloading as this. Are you opening the document in MS Excel or LibreOffice? I ran across an article saying that some versions of LibreOffice now have a different default char set than utf-8 and it has been causing some users some angst. I was only able to replicate the characters in your attachment if I set the char set to 'unicode'. MS Excel should have a similar char set selector when opening new files, unfortunately I was unable to test this at the moment. If you would like,  please check if switching this helps and let me know.

Thank you for all your help.

Kind regards,

Cassie

![char_encoding_utf8](https://cloud.githubusercontent.com/assets/10081573/17649736/db1360b2-61f0-11e6-9cca-70af7619ab22.PNG)
![char_encoding_unicode](https://cloud.githubusercontent.com/assets/10081573/17649737/dbc95d9a-61f0-11e6-9f2e-1f1acc371af6.PNG)

**Comment by cheinle** [comment]
Hi Morgan,

On the chance that you have already thought of what I mentioned a few moments ago, I have attached a new download.txt which can replace the contents of download.php if you would like. This should have a more explicit character setting. I also put this change in the github files if this is easier.

Kind regards,

Cassie
[download.txt](https://github.com/cheinle/NanoLIMS/files/417058/download.txt)
